loki/visitors/pretty_visualize.py

```diff
@@ -0,0 +1,349 @@
+        """
+        node_edge_info = self.__add_node(o, **kwargs)
+        kwargs["parent"] = o
+
+        node_edge_info.extend(self.visit(o.docstring, **kwargs))
+        node_edge_info.extend(self.visit(o.spec, **kwargs))
+        node_edge_info.extend(self.visit(o.body, **kwargs))
+        node_edge_info.extend(self.visit(o.members, **kwargs))
+
+        return node_edge_info
+
+    # Handler for AST base nodes
+    def visit_Comment(self, o, **kwargs):
+        """
+        Enables turning off comments.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information, or list of nothing.
+        """
+        if kwargs.get("show_comments"):
+            return self.visit_Node(o, **kwargs)
+        return [None]
+
+    visit_CommentBlock = visit_Comment
+
+    def visit_Node(self, o, **kwargs):
+        """
+        Add a :any:`Node`, mark parent and visit all children.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information
+        """
+        node_edge_info = self.__add_node(o, **kwargs)
+        kwargs["parent"] = o
+
+        node_edge_info.extend(self.visit_all(o.children, **kwargs))
+        return node_edge_info
+
+    def visit_Expression(self, o, **kwargs):
+        """
+        Dispatch routine to add nodes utilizing expression tree stringifier,
+        mark parent and stop.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information or list of nothing.
+        """
+        if kwargs.get("show_expressions"):
+            content = self.symgen(o)
+            parent = kwargs.get("parent")
+            return self.__add_node(o, label=content, parent=parent, shape="box")
+        return [None]
+
+    def visit_tuple(self, o, **kwargs):
+        """
+        Recurse for each item in the tuple.
+        """
+        return self.visit_all(o, **kwargs)
+
+    visit_list = visit_tuple
+
+    def visit_Conditional(self, o, **kwargs):
+        """
+        Add a :any:`Conditional`, mark parent and visit first body then else body.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information
+        """
+        parent = kwargs.get("parent")
+        label = self.symgen(o.condition)
+        node_edge_info = self.__add_node(o, label=label, parent=parent, shape="diamond")
+        kwargs["parent"] = o
+        node_edge_info.extend(self.visit_all(o.body, **kwargs))
+
+        if o.else_body:
+            node_edge_info.extend(self.visit_all(o.else_body, **kwargs))
+        return node_edge_info
+
+
+def pretty_visualize(ir, **kwargs):
```

Comment:
Given that you only expect a number of specific keys (filename, show_comments, show_expressions), I would prefer having them explicitly as optional arguments in the function signature instead of a plain `kwargs`

Reply:
parameters are now explicitly passed c72fe87359edd81a3f233e85478edf849f7b2c33